### PR TITLE
Bescka patch 6

### PR DESCRIPTION
**Pull request by bescka**
* Bugfix for content loading with github pages url
* Added custom themed prisma styling for code blocks
* Codewars problems updates in fn folder with testing framework
* Article on perfect squares
* Article on Recursion
* First commit for Algorithms Section

### CODE REVIEW

File: users/bescka/content/algorithms/queues.html
Comment (maxitect) [inline users/bescka/content/algorithms/queues.html:1]: This is really cool to see how classes and their methods can be built in Javascript, not explored that yet myself! Worth explaining what a class is or dropping a link somewhere for readers to find more information? It's a bit of a conceptual step from plain functions

File: users/bescka/content/algorithms/bubble_sort.html
[the remark this answers is not in the document]
Comment (bescka) [reply users/bescka/content/algorithms/bubble_sort.html:1]: Haha great :). Yes all the algos are in typescript just to be a bit more explicit. Nice spot you are right shouldn't have target it isn't a search algo. I've changed it in current branch thanks.

File: users/bescka/content/algorithms/recursion.html
[the remark this answers is not in the document]
Comment (bescka) [reply users/bescka/content/algorithms/recursion.html:1]: You can solve the problem with a db query. For example if i had a column with a JSON inside the cells like this: 
{
    "things": {
      "otherThings": [
        {
          "stuff": 42
        }
      ]
    }
  },
  {
    "things": {
      "otherThings": [
        {
          "stuff": 43
        }
      ]
    }
  }
]}

You can split out with something like this (postgres): 
SELECT json_array_elements_text(
           json_array_elements(json_column)->'things'->'otherThings'
       )::json->>'stuff' AS stuff_value
FROM example_table;

In the particular application I was using it for, using recursion was a nice way to flexibly traverse the nesting structure without having to write it out explicitly. It was also good for easily extending the data model using just the keys you want to look for. Tbh would probably make more sense in the context of a datamodel but wanted to keep focused on the algo itself.

File: users/bescka/content/articles/cuba_chups.html
[the remark this answers is not in the document]
Comment (bescka) [reply users/bescka/content/articles/cuba_chups.html:1]: works for me - will see if it works on pages after merge if not will update

**Comment by bescka** [comment]
Hey thanks for the review :). Yes for local you will need to start the server from bescka/. Will have a look if the pages integration works with the merge if not will open back up.

File: users/bescka/content/articles/cuba_chups.html
[the remark this answers is not in the document]
Comment (maxitect) [reply users/bescka/content/articles/cuba_chups.html:1]: Seems to work now on the deployed website!

**Comment by maxitect** [comment]
On the deployed site: 
It seems to be when you first load the website there is the screen that says "Error loading content." but then when you jump into a section using the menu it is resolved and everything works fine from that point on. It is a workaround but it works! Could just be my experience with my computer though.

**Pull request merged**